numba/tests/test_parfors.py

```diff
@@ -133,26 +157,35 @@ def copy_args(*args):
                 elif isinstance(x, numbers.Number):
                     new_args.append(x)
                 elif isinstance(x, tuple):
-                    new_args.append(x)
+                    new_args.append(copy.deepcopy(x))
                 elif isinstance(x, list):
                     new_args.append(x[:])
                 else:
                     raise ValueError('Unsupported argument type encountered')
             return tuple(new_args)
 
         # python result
-        py_expected = pyfunc(*copy_args(*args))
+        py_args = copy_args(*args)
+        py_expected = pyfunc(*py_args)
 
         # njit result
-        njit_output = cfunc.entry_point(*copy_args(*args))
+        njit_args = copy_args(*args)
+        njit_output = cfunc.entry_point(*njit_args)
 
         # parfor result
-        parfor_output = cpfunc.entry_point(*copy_args(*args))
-
-        np.testing.assert_almost_equal(njit_output, py_expected, **kwargs)
-        np.testing.assert_almost_equal(parfor_output, py_expected, **kwargs)
+        parfor_args = copy_args(*args)
+        parfor_output = cpfunc.entry_point(*parfor_args)
 
-        self.assertEqual(type(njit_output), type(parfor_output))
+        if check_args_for_equality is None:
+            np.testing.assert_almost_equal(njit_output, py_expected, **kwargs)
+            np.testing.assert_almost_equal(parfor_output, py_expected, **kwargs)
+            self.assertEqual(type(njit_output), type(parfor_output))
+        else:
+            assert(len(pyarg) == len(check_args_for_equality))
```

Comment:
```suggestion
            assert(len(py_args) == len(check_args_for_equality))
```
this should fix the unbound local problem